hack/Azure/setup/joinMC.sh

```diff
@@ -61,7 +67,6 @@ EOF
 # more information.
 echo "Retrieving image..."
 export REGISTRY="mcr.microsoft.com/aks/fleet"
-export FLEET_VERSION="${FLEET_VERSION:-$(curl "https://api.github.com/repos/Azure/fleet/tags" | jq -r '.[0].name')}"
```

Comment:
Not the best way to get the latest available tag, v0.10.9 seems to have been released and deleted. As of writing v0.10.8 is the latest available tag but the script grabbed v0.10.9 hence making it a user provided argument to reduce unforeseen errors 

![image](https://github.com/user-attachments/assets/e1d6822d-bbe9-4e9f-8ff1-1e99cf50bfe2)

Reply:
If we are changing this in this file, please change it for [createHubCluster.sh](https://github.com/Azure/fleet/blob/3915f0099ade7a57b726453028d3c27ef8db068e/hack/Azure/setup/createHubCluster.sh#L19). It will have the same issues.